templates/web/base/around/geolocate_link_icon.html

```diff
@@ -1,2 +1,2 @@
-<svg class="geolocate-link__icon" width="16" height="16" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M23.744.255a.874.874 0 01.18.974l-9.89 22.252a.874.874 0 01-1.603-.012l-3.568-8.332-8.334-3.57a.874.874 0 01-.01-1.602L22.77.075a.874.874 0 01.972.18h.002z" fill="currentColor" fill-rule="nonzero"/></svg>
-<svg class="geolocate-link__loading" width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><g fill="currentColor" fill-rule="nonzero"><path d="M16 8a8 8 0 11-8-8v2a6 6 0 106 6z"></path><circle cx="8" cy="1" r="1"></circle><circle cx="15" cy="8" r="1"></circle></g></svg>
+<svg class="geolocate-link__icon" width="16" height="16" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" role="img" aria-hidden="true"><path d="M23.744.255a.874.874 0 01.18.974l-9.89 22.252a.874.874 0 01-1.603-.012l-3.568-8.332-8.334-3.57a.874.874 0 01-.01-1.602L22.77.075a.874.874 0 01.972.18h.002z" fill="currentColor" fill-rule="nonzero"/></svg>
+<svg class="geolocate-link__loading" width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" role="img" aria-hidden="true"><g fill="currentColor" fill-rule="nonzero"><path d="M16 8a8 8 0 11-8-8v2a6 6 0 106 6z"></path><circle cx="8" cy="1" r="1"></circle><circle cx="15" cy="8" r="1"></circle></g></svg>
```

Comment:
I agree on the little icon, but the loading, we're going to need something other than hiding it, aren't we? https://stackoverflow.com/questions/38704467/how-to-label-a-loading-animation-for-wai-aria seems to provide a few different options, but I think geolocation.js will need some changes to improve this.

Reply:
mmm good point, however the loading state is so fast that most of the time the screen reader won't have the time to update the user, however while working on this I noticed related issue that we currently have, which is any error message, for example if the user declined the permission popup; the message "You declined..." doesn't get read. To solve this I added `aria-live="polite"` so whenever the message gets updated now it gets announced(I tried it with VoiceOver).

For the scenario when the user gets ask location permissions I noted that VoiceOver jumps straight away to the dialogue pop up, so this is already providing some context about what is happening the user. But I figured it was still worth it to include a message in the button in case some assistive devices behave differently.

I added a message when the location permission is granted, again this seems more a formality or for cases where the loading might be slow, but most of the time the screen reader won't have the time to read the message and we'll jump straight to the map page.